### PR TITLE
Implements Preserver#local_checksum_for in order to locally calculate the MD5 checksum before uploading preserved files to a Cloud Storage

### LINKED ISSUE #2897: Files uploaded to GCS should be validated via MD5
Unsure what the API for this should be like - Valkyrie doesn't really support this. The library will though:

http://googleapis.github.io/google-cloud-ruby/docs/google-cloud-storage/latest/Google/Cloud/Storage/Bucket.html#create_file-instance_method

### PR DESCRIPTION
**Pull request by jrgriffiniii**
Resolves #2897, but is blocked by https://github.com/samvera/valkyrie/pull/722.

### CODE REVIEW

**Comment by tpendragon** [comment]
Why did this require so many changes to the Preserver? The MD5 was already being calculated, I would have expected converting it to binary format and sending that would have been sufficient

File: app/services/preserver.rb [hunk 1]
```diff
@@ -98,13 +81,13 @@ def preserve_children
...
     def preserve_metadata
-      local_md5_checksum = local_checksum_for(file: temp_metadata_file.io)
+      local_checksum = metadata_node.checksum.first
 
       uploaded_file = storage_adapter.upload(
         file: temp_metadata_file.io,
         original_filename: metadata_node.label.first,
         resource: resource,
-        md5: local_md5_checksum
+        md5: local_checksum.md5
```
Comment (tpendragon) [inline app/services/preserver.rb]: You'll have to convert this to binary. I think it's `Base64.strict_encode64([local_checksum.md5].pack('H*'))`

Comment (jrgriffiniii) [reply app/services/preserver.rb]: No, I did not test against GCS, I can do this now.

Comment (jrgriffiniii) [reply app/services/preserver.rb]: I can confirm that `Base64.strict_encode64([local_checksum.md5].pack('H*'))` worked with a personal Google Cloud Storage Bucket.

**Comment by jrgriffiniii** [comment]
This is still blocked by the 2.0 release of `Valkyrie`, but improvements to these changes and the documentation are welcome please.

**Comment by jrgriffiniii** [comment]
This is no longer blocked by a 2.0 release of `valkyrie` (given https://github.com/samvera/valkyrie/releases/tag/v1.7.1), and following the completion of https://circleci.com/gh/pulibrary/figgy/17863?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link, I can squash the commits and offer this as a PR.

**Pull request merged**